numpy/lib/function_base.py

```diff
@@ -514,8 +514,7 @@ def average(a, axis=None, weights=None, returned=False):
         avg = a.mean(axis)
         scl = avg.dtype.type(a.size/avg.size)
     else:
-        a = a + 0.0
-        wgt = np.asarray(weights)
+        wgt = np.asarray(weights, dtype=float)
```

Comment:
unfortunately that won't work as result_type needs an ndarray as input, which is was asarray creates